### ISSUE TITLE
Finding temporary Pods project directory

### ISSUE THREAD

**Issue by liamogierussell** [CocoaPods/CocoaPods#3845]
Hey,

This is not really an issue but rather something iv found to be undocumented or possibly something I may be misunderstanding. I am just wondering is there any way to find the directory of the Pods project in the post install hook? I want to access the directory of the Pods project being generated as opposed to the Pods project that already exists in the Pods directory. I assume there is a temp directory where you generate the pods project and then in the install phase you replace the existing pods project with the newly generated one?

Thanks,
Liam.

**Comment by liamogierussell** [comment]
And just for context I am adding build phase scripts to each of the targets of the pods project and i need to iterate over the targets and build them in the post install hook (to generates a file i need when the project is opened).

**Comment by segiddins** [comment]
It'll be the sandbox root in the post install context.

**Comment by liamogierussell** [comment]
Hey, so when i do

```
post_install do |installer|
    
    puts installer.sandbox_root

```

This returns the directory of the existing Pods Project (i.e Users/MyUsername/MyProject/Pods) instead of a different directory where the Pods project being generated resides. Maybe I am wrong that when you build the cocoapods project that it is built in a seperate directory then copied over after the post install hook? Or maybe I am just accessing the directory incorrectly with installer.sandbox_root?

**Comment by segiddins** [comment]
The pods project is generated in place.

**Comment by liamogierussell** [comment]
That's the part that confuses me, so I run pod install and stop execution in the post install hook. At that point there is no pods project in the Pods directory, yet I am still given a reference to the project in the post install hook and can add build phases etc. Since it is generated in place shouldn't the project be in that directory in the post_install hook phase?

**Comment by segiddins** [comment]
It just might not be saved to disk yet?

**Comment by liamogierussell** [comment]
Ah ok ya probably just stored in a variable at that point and saved to disk in the integration phase. I'll have to try a different approach so. Thanks for your help segiddins.

**Comment by ogierussell** [comment]
I found a way to do this in the end, just putting it here for reference in case anyone needs to do run a post integration script in future. Basically my issue was that i needed to be able to change the pods project post integration and run xcodebuild on it. However there is no post_integration hook in the podfile. What I did was created a setup.sh file which had two lines

    pod install
    ruby myScript.sh

Then in myScript.sh I used xcodeproj to add the build phases, saved the project and then ran xcodebuild which was then building the targets with the correct run sctipts.
